src/handlers.js

```diff
@@ -87,18 +91,36 @@ export function taskParentGen(id)
     taskbar.appendChild(element);
     return element;
 }
-export function eventDispatcher(requestedAppId)
+export function eventDispatcher(requestedAppId,content)
 {
     let id=idGen(10);
     var isRunning=isAlreadyRunning(appRegistry[requestedAppId].processName)
-    eventHandler[id]=new runningTasks(id,appRegistry[requestedAppId]);
-    console.log(eventHandler[id].processIcon);
-    if(!isRunning)
-    {
-        var tPG=taskParentGen(id);
-        ReactDOM.render(<TaskbarIcon id={id} name={eventHandler[id].processName} location={`${eventHandler[id].iconData}`}/>,tPG)
+    //may add a if statement here 
+    if(requestedAppId==='editor'){
+        console.log(content)
+        eventHandler[id] = new runningTasks(id,appRegistry[requestedAppId],content)
+        if(!isRunning)
+        {
+            var tPG=taskParentGen(id);
+            console.log('here')
+            ReactDOM.render(<TaskbarIcon id={id} name={eventHandler[id].processName} location={`${eventHandler[id].iconData}`}/>,tPG)
+        }
+        ReactDOM.render(<Window id={id} source={eventHandler[id].Source}/>,parentGen(id));
```

Comment:
you can add something like data={content} here and in editor.js when the component is mounted you could set the editors code to be this state.data. Also we are probably increasing data transfer overheads by passing large chunk of text as props. But for now its fine. Later we can add APIs for fs operations that can be performed from any app. Try this way and let me know. Thanks

Reply:
I tried doing this before. It showed up the same way, blank.

Reply:
I tried passing a simple text in the prop for eg data=“hello” it did work (make sure you set the prop text onto ur state ) it works when i printed it on the console. So if you can make sure correct stuff is being passed around as a prop it would be awesome. Let me know what happens.

Reply:
Done 🚀